metaflow/runtime.py

```diff
@@ -1027,7 +1027,13 @@ def results(self):
     @property
     def finished_id(self):
         # note: id is not available before the task has finished
-        return (self.step, tuple(self.results["_foreach_stack"]))
+        # index already identifies the task within the foreach,
```

Comment:
why is this needed? Isn't the presence of the value backward compatible with the old code?

Reply:
for a node, we may want to quickly get the finish_id of the sibling node and check if they all finished (can check usage of `finished_id` within `_queue_task_join`). It is non-trivial to get value of the sibling, so we may as well not compare value at all (afterall, the index of the foreach already identifies the node).